st2common/st2common/logging/misc.py

```diff
@@ -43,12 +43,21 @@ def reopen_log_files(handlers):
 
         LOG.info('Re-opening log file "%s" with mode "%s"\n' %
                  (handler.baseFilename, handler.mode))
+
         try:
             handler.acquire()
```

Comment:
Another option would be to wrap acquire in `try / except`. This way we could avoid nasty if below.